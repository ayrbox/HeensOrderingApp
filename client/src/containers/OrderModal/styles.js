const styles = theme => ({
  modalContainer: {
    height: '100%',
    width: '250px',
    padding: `${theme.spacing.unit * 4}px`,
  },
  orderButton: {
    marginBottom: `${theme.spacing.unit}px`,
  },
  fullWidth: {
    width: '100%',
  },
});

export default styles;
