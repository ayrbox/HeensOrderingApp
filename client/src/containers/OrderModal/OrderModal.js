import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import Drawer from '@material-ui/core/Drawer';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import { withStyles } from '@material-ui/core/styles';

import styles from './styles';

const OrderModal = ({
  classes,
  isOpenOrderModal,
  resetOrder,
  setOrderType,
  history,
}) => {
  const [modalStep, setModalStep] = useState(0);

  const handleOrderType = (e, orderType) => {
    e.preventDefault();
    setOrderType(orderType);
    history.push('/orders/new');
    resetOrder();
  };

  return (
    <Drawer
      open={isOpenOrderModal}
      anchor="right"
      onClose={resetOrder}
    >
      <Grid
        container
        className={classes.modalContainer}
        direction="column"
        justify="space-between"
        alignItems="center"
      >
        <Grid
          item
          className={classes.fullWidth}
        >
          <Grid
            container
            direction="row"
            justify="space-between"
          >
            {modalStep === 1 ? (
              <IconButton
                aria-label="Close"
                onClick={(e) => {
                  e.preventDefault();
                  setModalStep(0);
                }}
              >
                <ArrowBackIcon fontSize="small" />
              </IconButton>
            ) : <div />}
            <IconButton
              aria-label="Close"
              onClick={resetOrder}
            >
              <CloseIcon fontSize="small" />
            </IconButton>
          </Grid>
        </Grid>
        {modalStep === 0 && (
          <Grid item>
            <Button
              variant="contained"
              color="primary"
              className={classes.orderButton}
              size="large"
              fullWidth
              onClick={e => handleOrderType(e, 'table')}
            >
              EAT IN
            </Button>
            <Button
              variant="outlined"
              color="secondary"
              className={classes.orderButton}
              size="large"
              fullWidth
              onClick={(e) => {
                e.preventDefault();
                setModalStep(1);
              }}
            >
              TAKE AWAY
            </Button>
          </Grid>
        )}
        {modalStep === 1 && (
          <Grid item>
            <Button
              variant="contained"
              color="primary"
              className={classes.orderButton}
              size="large"
              fullWidth
              onClick={e => handleOrderType(e, 'delivery')}
            >
              DELIVERY
            </Button>
            <Button
              variant="outlined"
              color="secondary"
              className={classes.orderButton}
              size="large"
              fullWidth
              onClick={e => handleOrderType(e, 'collection')}
            >
              COLLECTION
            </Button>
          </Grid>
        )}
        <Grid item />
      </Grid>
    </Drawer>
  );
};

OrderModal.propTypes = {
  classes: PropTypes.shape().isRequired,
  isOpenOrderModal: PropTypes.bool.isRequired,
  resetOrder: PropTypes.func.isRequired,
  setOrderType: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
};

export default withStyles(styles)(withRouter(OrderModal));
