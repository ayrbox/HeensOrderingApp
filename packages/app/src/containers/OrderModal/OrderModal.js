import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';
import Drawer from '@material-ui/core/Drawer';
import Grid from '@material-ui/core/Grid';
import Button from '@material-ui/core/Button';
import IconButton from '@material-ui/core/IconButton';
import CloseIcon from '@material-ui/icons/Close';
import ArrowBackIcon from '@material-ui/icons/ArrowBack';
import { withStyles } from '@material-ui/core/styles';

import styles from './styles';

const OrderModal = ({
  classes,
  isOpenOrderModal,
  closeOrderPane,
  setOrderType,
  history,
}) => {
  const [modalStep, setModalStep] = useState(0);

  const handleOrderType = (e, orderType) => {
    e.preventDefault();
    setOrderType(orderType);
    history.push('/orders/new');
    closeOrderPane();
  };

  return (
    <Drawer
      open={isOpenOrderModal}
      anchor="right"
      onClose={closeOrderPane}
      data-testid="order-drawer"
    >
      <Grid
        container
        className={classes.modalContainer}
        direction="column"
        justify="space-between"
        alignItems="center"
      >
        <Grid
          item
          className={classes.fullWidth}
        >
          <Grid
            container
            direction="row"
            justify="space-between"
          >
            {modalStep === 1 ? (
              <IconButton
                aria-label="Close"
                onClick={(e) => {
                  e.preventDefault();
                  setModalStep(0);
                }}
                data-testid="button-back"
              >
                <ArrowBackIcon fontSize="small" />
              </IconButton>
            ) : <div />}
            <IconButton
              aria-label="Close"
              onClick={closeOrderPane}
              data-testid="button-close"
            >
              <CloseIcon fontSize="small" />
            </IconButton>
          </Grid>
        </Grid>
        {modalStep === 0 && (
          <Grid item>
            <Button
              variant="contained"
              color="primary"
              className={classes.orderButton}
              size="large"
              fullWidth
              onClick={e => handleOrderType(e, 'table')}
              data-testid="button-eatin"
            >
              EAT IN
            </Button>
            <Button
              variant="outlined"
              color="secondary"
              className={classes.orderButton}
              size="large"
              fullWidth
              onClick={(e) => {
                e.preventDefault();
                setModalStep(1);
              }}
              data-testid="button-takeaway"
            >
              TAKE AWAY
            </Button>
          </Grid>
        )}
        {modalStep === 1 && (
          <Grid item>
            <Button
              variant="contained"
              color="primary"
              className={classes.orderButton}
              size="large"
              fullWidth
              onClick={e => handleOrderType(e, 'delivery')}
              data-testid="button-delivery"
            >
              DELIVERY
            </Button>
            <Button
              variant="outlined"
              color="secondary"
              className={classes.orderButton}
              size="large"
              fullWidth
              onClick={e => handleOrderType(e, 'collection')}
              data-testid="button-collection"
            >
              COLLECTION
            </Button>
          </Grid>
        )}
        <Grid item />
      </Grid>
    </Drawer>
  );
};

OrderModal.propTypes = {
  classes: PropTypes.shape().isRequired,
  isOpenOrderModal: PropTypes.bool.isRequired,
  closeOrderPane: PropTypes.func.isRequired,
  setOrderType: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
};

export default withStyles(styles)(withRouter(OrderModal));
