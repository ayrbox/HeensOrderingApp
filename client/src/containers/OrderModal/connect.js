import { connect } from 'react-redux';
import OrderModal from './OrderModal';

import { closeOrderModal } from '../../actions/orderActions';

const mapState = state => ({
  isOpenOrderModal: state.orders.isOpenOrderModal,
});

export default connect(mapState, {
  closeOrderModal,
})(OrderModal);
