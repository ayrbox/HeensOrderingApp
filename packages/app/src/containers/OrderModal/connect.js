import { connect } from 'react-redux';
import OrderModal from './OrderModal';

import { closeOrderPane, setOrderType } from '../../actions/newOrderActions';

const mapState = state => ({
  isOpenOrderModal: state.newOrder.openNewOrderPane,
});

export default connect(mapState, {
  closeOrderPane,
  setOrderType,
})(OrderModal);
