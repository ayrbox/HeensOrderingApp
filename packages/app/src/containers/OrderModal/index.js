import OrderModal from './connect';

export default OrderModal;
