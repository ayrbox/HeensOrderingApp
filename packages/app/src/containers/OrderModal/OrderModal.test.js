import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import {
  render,
  fireEvent,
  cleanup,
  waitForElement,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import OrderModal from './OrderModal';

const mockCloseOrderPane = jest.fn();
const mockSetOrderType = jest.fn();


const makeOrderModal = () => render(
  <Router>
    <OrderModal
      isOpenOrderModal
      closeOrderPane={mockCloseOrderPane}
      setOrderType={mockSetOrderType}
    />
  </Router>,
);

describe('<OrderModal />', () => {
  afterEach(() => {
    mockCloseOrderPane.mockClear();
    mockSetOrderType.mockClear();
    cleanup();
  });

  describe('when close buttin is clicked', () => {
    it('should trigger close func', () => {
      const { getByTestId } = makeOrderModal();
      getByTestId('order-drawer');

      fireEvent.click(getByTestId(/button-close/));
      expect(mockCloseOrderPane).toHaveBeenCalled();
    });
  });

  describe('when eat-in button is clicked', () => {
    it('should trigger set order and close', () => {
      const {
        container,
        getByTestId,
      } = makeOrderModal();
      getByTestId('order-drawer');

      // Eat In button triggers set order and close
      fireEvent.click(getByTestId('button-eatin'));
      expect(mockSetOrderType).toHaveBeenCalledWith('table');
      expect(mockCloseOrderPane).toHaveBeenCalled();

      expect(container).toMatchSnapshot();
    });
  });

  describe('when takeaway button is clicked', () => {
    it('should reveal delivery and collection button', async () => {
      const { container, getByTestId } = makeOrderModal();
      getByTestId(/order-drawer/);

      // Takeway button triggers next step
      fireEvent.click(getByTestId(/button-takeaway/));
      await waitForElement(() => [
        getByTestId(/button-delivery/),
        getByTestId(/button-collection/),
        getByTestId(/button-back/),
      ]);
      expect(container).toMatchSnapshot();
    });
  });


  describe('when delivery/collection buttons clicked', () => {
    it('should trigger set order with delivery', async () => {
      const { getByTestId } = makeOrderModal();
      fireEvent.click(getByTestId(/button-takeaway/));

      const btnDelivery = await waitForElement(() => getByTestId(/button-delivery/));

      fireEvent.click(btnDelivery);
      expect(mockSetOrderType).toHaveBeenCalledWith('delivery');
      expect(mockCloseOrderPane).toHaveBeenCalled();
    });

    it('should trigger set order with collection', async () => {
      const { getByTestId } = makeOrderModal();
      fireEvent.click(getByTestId(/button-takeaway/));

      const btnCollection = await waitForElement(() => getByTestId(/button-collection/));

      fireEvent.click(btnCollection);
      expect(mockSetOrderType).toHaveBeenCalledWith('collection');
      expect(mockCloseOrderPane).toHaveBeenCalled();
    });

    it('should trigger screen back to previous', async () => {
      const { getByTestId } = makeOrderModal();
      fireEvent.click(getByTestId(/button-takeaway/));

      const btnBack = await waitForElement(() => getByTestId(/button-back/));

      fireEvent.click(btnBack);
      await waitForElement(() => [
        getByTestId(/button-eatin/),
        getByTestId(/button-takeaway/),
      ]);
    });
  });
});
