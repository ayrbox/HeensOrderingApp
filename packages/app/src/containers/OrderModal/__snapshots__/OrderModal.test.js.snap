// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderModal /> should render correctly 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`<OrderModal /> when eat in button is clicked should trigger set order and close 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`<OrderModal /> when eat-in button is clicked should trigger set order and close 1`] = `
<div
  aria-hidden="true"
/>
`;

exports[`<OrderModal /> when takeaway button is clicked should reveal delivery and collection button 1`] = `
<div
  aria-hidden="true"
/>
`;
